src/Forms/GridField/GridFieldFilterHeader.php

```diff
@@ -456,4 +463,32 @@ public function getHTMLFragments($gridField)
             )
         ];
     }
+
+    /**
+     * Get the text that will be used as a placeholder in the search field.
+     *
+     * @param object $obj An instance of the class that will be searched against.
+     * If getPlaceHolderText is empty, this object will be used to build the placeholder
+     * e.g. 'Search "My Data Object"'
+     */
+    private function getPlaceHolder(object $obj): string
+    {
+        $placeholder = $this->getPlaceHolderText();
+        if (!empty($placeholder)) {
+            return $placeholder;
+        }
+        if ($obj) {
+            return _t(__CLASS__ . '.Search', 'Search "{name}"', ['name' => $this->getTitle($obj)]);
+        }
+        return _t(__CLASS__ . '.Search_Default', 'Search');
+    }
+
+    private function getTitle(object $inst): string
+    {
+        if (ClassInfo::hasMethod($inst, 'i18n_plural_name')) {
+            return $inst->i18n_plural_name();
+        }
+
+        return ClassInfo::shortName($inst);
+    }
```

Comment:
Note I've removed the code that relied on `$gridField->Title` because that was effectively dead code.
I've removed the corresponding argument for `$gridField` from the method signature since this is a private method.